### PR TITLE
Revert /tmp to tmpfs

### PR DESCRIPTION
**Pull request by edsantiago**
Podman *really* needs /tmp to be tmpfs, to detect and
handle reboots. Although there are (at this time) no
reboots involved in CI testing, it's still important
for CI hosts to reflect something close to a real-world
environment. And, there is work underway to check /tmp:

  https://github.com/containers/podman/pull/22141

This PR removes special-case Fedora code that was
disabling a tmpfs /tmp mount. History dates back to
PR #30 back in 2020.

Some of the image-build code in this repo performs
reboots and relies on persistent tmp files, so you'll
note a flurry of /tmp -> /var/tmp changes.

And, as a drive-by, document the Windows Chocolatey
install command. Link to Best Practices, and explain
why we disregard some of those.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
Retried three times, it's not a flake:
```
�[0;32m    win-server-wsl:  - golang - golang not installed. The package was not found with the source(s) listed.�[0m
```
Any Windows experts know what's going on with Chocolaty and Golang?

**Comment by edsantiago** [comment]
The git flake resolved itself, now it's back to golang failing. There's one difference which I think might be important, having to do with versions vs no-versions:

[git flake](https://cirrus-ci.com/task/5638422398763008?logs=main#L107):
```
win-server-wsl: Unable to resolve dependency: \
    Unable to find a version of 'git.install' that is compatible with 'git 2.44.0 constraint: \
    git.install (= 2.44.0)'.
```
[go failure](https://cirrus-ci.com/task/4581630864523264?logs=main#L173):
```
win-server-wsl: golang not installed. The package was not found with the source(s) listed.
(no version string at all)
```
Maybe the git flake was like a dnf cache delay, where packages just weren't refreshed in repos? `git 2.44` is a month old, but maybe it only just got built for windows today. And maybe the golang problem is that golang just got completely removed from the chocolate thing?

**Comment by edsantiago** [comment]
@containers/podman-maintainers I give up. This needs a Windows expert.

**Comment by jwhonce** [comment]
@l0rd Any ideas?

**Comment by edsantiago** [comment]
YAY! Passed after enough retries!

- 20240326t173017z-f39f38d13
- 20240320t153921z-f39f38d13 &#x21D1; (built in #337)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13              | 38-1.6          | 39-1.5          | ?               | 41-0            |
|               |                |                |                |                |                |
|*kernel*       | 6.7.9-2         | 6.7.9-100       | 6.7.10-200      | 6.7.9-200       | 6.9.0-0.rc0.20240322git8e938e398669.14 |
|               |                |                |                |                | 6.9.0-0.rc0.20240318gitf6cef5f8c37f.10&nbsp;&#x21D1; |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-1        | 1.10.3-1        | 1.10.3-3        |
|               |                |                |                |                | 1.10.3-2&nbsp;&#x21D1; |
|*buildah*      | 1.33.5+ds1-4+b1 | 1.34.0-1        | 1.35.1-1        | 1.35.0-1        | 1.35.0-1        |
|               | 1.33.5+ds1-4&nbsp;&#x21D1; |                |                |                |                |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       |
|               |                | 2.228.1-1&nbsp;&#x21D1; |                |                |                |
|*containers-common*| ?               | 1-89            | 1-99            | 1-99            | 0.58.0-4        |
|               |                |                |                |                | 0.58.0-1&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.18-1          | 3.19-2          | 3.19-2          | 3.19-4          |
|               |                |                |                |                |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        |
|               |                |                |                |                |                |
|*golang*       | 2:1.22~3        | 1.21.8-1        | 1.21.8-1        | 1.21.8-1        | 1.22.1-1        |
|               |                |                |                |                |                |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-1         | 0.7.3-1         | 0.7.3-2         |
|               |                |                |                |                |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.93-2          | 7.94-1          | 7.94-1          | 7.94-1          |
|               |                |                |                |                |                |
|*passt*        | 2024-03-20      | 2024-03-20      | 2024-03-20      | 2024-03-20      | 2024-03-20      |
|               | 2024-02-20&nbsp;&#x21D1; |                |                |                |                |
|*podman*       | 4.9.3+ds1-1+b1  | 4.9.3-2         | 4.9.3-1         | 4.9.3-1         | 5.0.0~rc6-2     |
|               | 4.9.3+ds1-1&nbsp;&#x21D1; |                |                |                |                |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-1        | 1.1.12-1        | 1.1.12-3        |
|               |                |                |                |                |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.14.2-1        | 1.14.2-2        |
|               | 1.13.3+ds1-2&nbsp;&#x21D1; | 1.14.2-1&nbsp;&#x21D1; |                |                |                |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-1         | 1.2.2-1         | 1.2.2-2         |
|               |                |                |                |                |                |
|*systemd*      | 255.4-1+b1      | 253.17-1        | 254.10-1        | 254.10-1        | 255.4-1         |
|               |                | 253.15-2&nbsp;&#x21D1; |                |                |                |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.34-8          | 1.35-2          | 1.35-2          | 1.35-3          |
|               |                |                |                |                |                |

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by l0rd** [comment]
> @l0rd Any ideas?

Packages [golang](https://community.chocolatey.org/packages/golang) and [git](https://community.chocolatey.org/packages/git) are available from the default source (`community.chocolatey.org`) and the [community services were operational yesterday](https://status.chocolatey.org/). [The troubleshooting guide](https://docs.chocolatey.org/en-us/troubleshooting#package-not-installed.the-package-was-not-found-with-the-sources-listed) suggests to check the source but we use the default one so that doesn't look to be the problem. Adding `--debug --verbose --noop` to the `choco install` command may help to get more information next time.

Anyway [chocolately scripting best practices](https://docs.chocolatey.org/en-us/choco/commands/#scripting-integration-best-practices-style-guide) suggest to use a slightly different command to install packages:

```diff
-- choco install -y --allow-downgrade --execution-timeout=300 $pkg
++ choco upgrade $pkg1 -y --source="'https://community.chocolatey.org/api/v2'" --allow-downgrade --execution-timeout=300
```

I don't think that this has anything to do with the problems above though.

**Comment by edsantiago** [comment]
/hold

Let's not merge until we see if this works. My two Podman PRs with these VMs have stuck in CI, twice. I don't know if it's a Cirrus problem or a problem with the VMs.

**Comment by edsantiago** [comment]
@cevich there's something broken somewhere. "Build for debian" is consistently [hanging](https://cirrus-ci.com/task/5611127002890240), not even getting started. So is [Validate f39](https://cirrus-ci.com/task/6737026909732864). AFAICS there is no indication of why they won't run. How can I debug this?

**Comment by cevich** [comment]
> How can I debug this?

They're stuck in 'scheduled' mode.  Off-hand, these are container-based Cirrus-CI compute-based tasks.  We're not out of credits :thinking:  It's running now, so only thing I can think of is: Some kind of Cirrus-CI hiccup.

**Comment by edsantiago** [comment]
Not sure what you mean by "it's running now". Maybe you're referring to "this PR, the /tmp-tmpfs one in automation_images, is running now"? If so, that has nothing to do with anything. The not-running thing is two podman PRs using images I built yesterday. The running-now thing is maybe this PR (the one on which I'm commenting), which I re-pushed in desperation. Maybe yesterday's VMs didn't work because of the Windows outage. I dunno. I'll see what happens in a few minutes.

**Comment by edsantiago** [comment]
- 20240327t142743z-f39f38d13
- 20240102t155643z-f39f38d13 &#x21D1;

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13              | 38-1.6          | 39-1.5          | ?               | 41-0            |
|               |                |                |                |                | 40-0&nbsp;&#x21D1; |
|*kernel*       | 6.7.9-2         | 6.7.10-100      | 6.7.10-200      | 6.7.10-200      | 6.9.0-0.rc1.17  |
|               |                |                |                |                |                |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                | 1.9.0-1&nbsp;&#x21D1; | 1.9.0-1&nbsp;&#x21D1; | 1.9.0-1&nbsp;&#x21D1; |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-1        | 1.10.3-1        | 1.10.3-3        |
|               |                |                | 1.9.0-1&nbsp;&#x21D1; | 1.9.0-1&nbsp;&#x21D1; | 1.9.0-1&nbsp;&#x21D1; |
|*buildah*      | 1.33.7+ds1-1    | 1.34.0-1        | 1.35.1-1        | 1.35.0-1        | 1.35.0-1        |
|               | 1.32.2+ds1-1&nbsp;&#x21D1; | 1.33.2-1&nbsp;&#x21D1; | 1.33.2-1&nbsp;&#x21D1; | 1.33.2-1&nbsp;&#x21D1; | 1.33.2-1&nbsp;&#x21D1; |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               | 2.1.6+ds1-1&nbsp;&#x21D1; | 2.1.8-2&nbsp;&#x21D1; | 2.1.8-2&nbsp;&#x21D1; | 2.1.8-2&nbsp;&#x21D1; | 2.1.8-2&nbsp;&#x21D1; |
|*container-selinux*| ?               | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       |
|               |                | 2.226.0-1&nbsp;&#x21D1; | 2.226.0-1&nbsp;&#x21D1; | 2.226.0-1&nbsp;&#x21D1; | 2.226.0-1&nbsp;&#x21D1; |
|*containers-common*| ?               | 1-89            | 1-99            | 1-99            | 0.58.0-5        |
|               |                |                | 1-95&nbsp;&#x21D1; | 1-95&nbsp;&#x21D1; | 1-101&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.18-1          | 3.19-2          | 3.19-2          | 3.19-4          |
|               |                |                |                |                | 3.19-2&nbsp;&#x21D1; |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        |
|               | 1.12-1&nbsp;&#x21D1; | 1.12-1&nbsp;&#x21D1; | 1.12-1&nbsp;&#x21D1; | 1.12-1&nbsp;&#x21D1; | 1.12-1&nbsp;&#x21D1; |
|*golang*       | 2:1.22~3        | 1.21.8-1        | 1.21.8-1        | 1.21.8-1        | 1.22.1-4        |
|               | 2:1.21~2&nbsp;&#x21D1; | 1.20.12-1&nbsp;&#x21D1; | 1.21.5-1&nbsp;&#x21D1; | 1.21.5-1&nbsp;&#x21D1; | 1.21.5-1&nbsp;&#x21D1; |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-1         | 0.7.3-1         | 0.7.3-2         |
|               |                | 0.7.1-1&nbsp;&#x21D1; | 0.7.1-1&nbsp;&#x21D1; | 0.7.1-1&nbsp;&#x21D1; | 0.7.1-1&nbsp;&#x21D1; |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.93-2          | 7.94-1          | 7.94-1          | 7.94-1          |
|               | 7.94+git20230807.3be01efb1+dfsg-2&nbsp;&#x21D1; |                | 7.93-3&nbsp;&#x21D1; | 7.93-3&nbsp;&#x21D1; | 7.93-4&nbsp;&#x21D1; |
|*passt*        | 2024-03-26      | 2024-03-20      | 2024-03-26      | 2024-03-20      | 2024-03-26      |
|               | 2023-12-04&nbsp;&#x21D1; | 2023-12-04&nbsp;&#x21D1; |                | 2023-12-04&nbsp;&#x21D1; | 2023-12-04&nbsp;&#x21D1; |
|*podman*       | 4.9.3+ds1-1+b1  | 4.9.3-2         | 4.9.4-1         | 4.9.3-1         | 5.0.0-1         |
|               | 4.7.2+ds1-2&nbsp;&#x21D1; | 4.7.2-1&nbsp;&#x21D1; | 4.8.2-1&nbsp;&#x21D1; | 4.8.1-1&nbsp;&#x21D1; | 4.8.1-1&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-1        | 1.1.12-1        | 1.1.12-3        |
|               | 1.1.10+ds1-1&nbsp;&#x21D1; | 1.1.8-1&nbsp;&#x21D1; | 1.1.8-1&nbsp;&#x21D1; | 1.1.8-1&nbsp;&#x21D1; | 1.1.9-1&nbsp;&#x21D1; |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.14.2-1        | 1.14.2-2        |
|               | 1.13.3+ds1-2&nbsp;&#x21D1; | 1.14.0-1&nbsp;&#x21D1; | 1.14.0-1&nbsp;&#x21D1; | 1.14.0-1&nbsp;&#x21D1; | 1.14.0-1&nbsp;&#x21D1; |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-1         | 1.2.2-1         | 1.2.2-2         |
|               | 1.2.1-1&nbsp;&#x21D1; |                |                |                | 1.2.2-1&nbsp;&#x21D1; |
|*systemd*      | 255.4-1+b1      | 253.17-1        | 254.10-1        | 254.10-1        | 255.4-1         |
|               | 255.2-3&nbsp;&#x21D1; |                |                |                | 255.1-1&nbsp;&#x21D1; |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.34-8          | 1.35-2          | 1.35-2          | 1.35-3          |
|               | 1.35+dfsg-2&nbsp;&#x21D1; |                |                |                | 1.35-2&nbsp;&#x21D1; |

**Comment by cevich** [comment]
> Not sure what you mean by "it's running now"

Oh my bad, I assumed those links were tasks in this repo.  I see they're in podman.  I just saw a re-push and now the replacement tasks are running over there.  This still seems like it was some temporary Cirrus-infra. hiccup.

**Comment by cevich** [comment]
BTW: I would strongly recommend testing these images beyond just podman CI.  Esp. `buildah` and `skopeo`.

**Comment by edsantiago** [comment]
@cevich nope, still [broken](https://cirrus-ci.com/task/5052720325853184). (That link right is to a podman debian-13 job, it is entirely gray, nothing happening on it whatsoever). Help please.

**Comment by cevich** [comment]
> nope, still [broken](https://cirrus-ci.com/task/5052720325853184).

Gah!  Hmmmm :thinking:

**Comment by edsantiago** [comment]
Debian grub changed. Not by much, but apparently enough to cause a boot failure.

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*grub2-common* | 2.12-1+b1       | 2.06-116        | 2.06-118        | 2.06-118        | 2.06-119        |
|               | 2.12~rc1-12&nbsp;&#x21D1; |                |                |                | 2.06-110&nbsp;&#x21D1; |

**Comment by cevich** [comment]
Cause (not root): Boot failure

```

[0m[30m[47mWelcome to GRUB!
 
 
[0m[37m[40m[0m[30m[40m[2J[01;01H[0m[37m[40m[H[J[1;1Herror: file `/boot/grub/x86_64-efi/bli.mod' not found.
 
error: file `/boot/grub/x86_64-efi/bli.mod' not found.
```

IIRC, we're using a custom `/usr/bin/version_find_latest` for debian (see `base_images/debian_base-setup.sh`), could that somehow be involved?

Clearly the base-image is bootable (since cache-image stage is using it and not failing).  So it must be something going wrong there.  Anything jump out at you in the build logs?

File: base_images/debian_base-setup.sh
Comment (cevich) [inline base_images/debian_base-setup.sh:153]: (Non-blocking) Did you happen to find anything online about this, that might be good to include in the comment?

I'm considering a frustrated "not Ed" maintainer coming along with a PR and not really knowing what to do with this timebomb (or any of them, really).  Would you consider (now or later) including comments on how to verify these?  I'm fine if that's simply: "Ask Ed" or a more complete/complex action-item.  Just something so people aren't left "hanging", wondering what to do.

File: cache_images/cloud.yml
Comment (cevich) [inline cache_images/cloud.yml:186]: Actually, this will (I think) have a knock-on effect:

All CI VM images will include a copy of the instructions/scripts used to build them.  I just checked, the `finalize()` functions don't touch `/var/tmp/`, only `/tmp`.

File: lib.sh
Comment (cevich) [inline lib.sh:303]: Maybe consider not doing this?  Could be handy for someone doing archaeology on an old release-branch CI VM wondering how it was built.

File: base_images/debian_base-setup.sh
Comment (edsantiago) [reply base_images/debian_base-setup.sh:153]: I found nothing, and didn't even bother looking. We already know for a fact that debian and google do not play well together (see the other grub timebomb); it doesn't seem useful to figure out why.

What _does_ bother me a lot is that there's no way (that I know of) to see if there's an update to the buggy version. Like, the `tar` block, maybe debian has fixed the bug and pushed a new version, but there's not really any way to look at the logs and see "oh hey there's a version beyond the timebomed one, let's see if that fixes things". I've just added a TODO item for myself to see if there's a debian way to block one specific version, instead of blocking all upgrades.

Anyhow, [IT WORKED!](https://cirrus-ci.com/task/5003396384555008). Now to see what else fails. If it passes podman and buildah CI, I'll address your other (very welcome, thank you) feedback.

File: lib.sh
Comment (edsantiago) [reply lib.sh]: I interpret "not doing this" as "undoing the added /var/tmp". So done.

File: cache_images/cloud.yml
Comment (edsantiago) [reply cache_images/cloud.yml:186]: I don't understand this comment but I think it is addressed by my "undo the rm /var/tmp/auto*" push just now. If it doesn't, please let me know.

File: base_images/debian_base-setup.sh
Comment (cevich) [reply base_images/debian_base-setup.sh]: Yay :tada: 

Renovate would be the tool for watching versions, it can accept a regex search pattern to locate the necessary "in use" version details.  The challenge is on the back-end, getting the "latest" versions of things that are available.  There's [a versioning module called repology](https://docs.renovatebot.com/modules/datasource/repology/), and it may work well for Debian. But the last time I tried, it fell flat on everything Fedora related :disappointed:  Anyway, there may be opportunities in this space, given the time/effort required to configure it.  Bother me later if you think this is really important and should be pursued further.

File: cache_images/cloud.yml
Comment (cevich) [reply cache_images/cloud.yml:186]: Ya, you got it.

File: base_images/debian_base-setup.sh
Comment (edsantiago) [reply base_images/debian_base-setup.sh]: Nah, already solved it, with a `/etc/apt` thing that failed because I forgot to SUDO and I'm not going to bother re-pushing today anyway.

**Comment by edsantiago** [comment]
I have reinstated the `rm -rf /var/tmp/automation_images` because, without it, [all cache jobs fail](https://cirrus-ci.com/task/5204043834851328?logs=main#L29):
```
==> fedora: Uploading /tmp/cirrus-ci-build/ => /var/tmp/automation_images
==> fedora: Upload failed: scp: /var/tmp/automation_images/SECURITY.md: Permission denied
```

**Comment by edsantiago** [comment]
Oh look, there's a new pasta (03-26) (only on some hosts)

- 20240328t122402z-f39f38d13
- 20240320t153921z-f39f38d13 &#x21D1; (built in #337)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13              | 38-1.6          | 39-1.5          | ?               | 41-0            |
|               |                |                |                |                |                |
|*kernel*       | 6.7.9-2         | 6.7.10-100      | 6.7.10-200      | 6.7.10-200      | 6.9.0-0.rc1.20240326git928a87efa423.17 |
|               |                | 6.7.9-100&nbsp;&#x21D1; |                | 6.7.9-200&nbsp;&#x21D1; | 6.9.0-0.rc0.20240318gitf6cef5f8c37f.10&nbsp;&#x21D1; |
|*grub2-common* | 2.06-13+deb12u1 | 2.06-116        | 2.06-118        | 2.06-118        | 2.06-119        |
|               | 2.12-1&nbsp;&#x21D1; | 2.06-114&nbsp;&#x21D1; |                |                |                |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-1        | 1.10.3-1        | 1.10.3-3        |
|               |                |                |                |                | 1.10.3-2&nbsp;&#x21D1; |
|*buildah*      | 1.33.7+ds1-1    | 1.35.1-1        | 1.35.2-1        | 1.35.0-1        | 1.35.0-1        |
|               | 1.33.5+ds1-4&nbsp;&#x21D1; | 1.34.0-1&nbsp;&#x21D1; | 1.35.1-1&nbsp;&#x21D1; |                |                |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       |
|               |                | 2.228.1-1&nbsp;&#x21D1; |                |                |                |
|*containers-common*| ?               | 1-89            | 1-99            | 1-99            | 0.58.0-5        |
|               |                |                |                |                | 0.58.0-1&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.18-1          | 3.19-2          | 3.19-2          | 3.19-4          |
|               |                |                |                |                |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        |
|               |                |                |                |                |                |
|*golang*       | 2:1.22~3        | 1.21.8-1        | 1.21.8-1        | 1.21.8-1        | 1.22.1-4        |
|               |                |                |                |                | 1.22.1-1&nbsp;&#x21D1; |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-1         | 0.7.3-1         | 0.7.3-2         |
|               |                |                |                |                |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.93-2          | 7.94-1          | 7.94-1          | 7.94-1          |
|               |                |                |                |                |                |
|*passt*        | 2024-03-26      | 2024-03-20      | 2024-03-26      | 2024-03-20      | 2024-03-26      |
|               | 2024-02-20&nbsp;&#x21D1; |                | 2024-03-20&nbsp;&#x21D1; |                | 2024-03-20&nbsp;&#x21D1; |
|*podman*       | 4.9.3+ds1-1+b1  | 4.9.3-2         | 4.9.4-1         | 4.9.3-1         | 5.0.0-1         |
|               | 4.9.3+ds1-1&nbsp;&#x21D1; |                | 4.9.3-1&nbsp;&#x21D1; |                | 5.0.0~rc6-2&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-1        | 1.1.12-1        | 1.1.12-3        |
|               |                |                |                |                |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.14.2-1        | 1.14.2-2        |
|               | 1.13.3+ds1-2&nbsp;&#x21D1; | 1.14.2-1&nbsp;&#x21D1; |                |                |                |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-1         | 1.2.2-1         | 1.2.2-2         |
|               |                |                |                |                |                |
|*systemd*      | 255.4-1+b1      | 253.17-1        | 254.10-1        | 254.10-1        | 255.4-1         |
|               |                | 253.15-2&nbsp;&#x21D1; |                |                |                |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.34-8          | 1.35-2          | 1.35-2          | 1.35-3          |
|               |                |                |                |                |                |

File: base_images/debian_base-setup.sh
Comment (cevich) [reply base_images/debian_base-setup.sh]: I always forget the `$SUDO` also :frowning_face:

**Comment by edsantiago** [comment]
Moved to draft until #342 merges.

**Comment by edsantiago** [comment]
- 20240401t112841z-f39f38d13
- 20240328t140922z-f39f38d13 &#x21D1; (built in #342)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13              | 38-1.6          | 39-1.5          | ?               | 41-0            |
|               |                |                |                |                |                |
|*kernel*       | 6.7.9-2         | 6.7.10-100      | 6.7.11-200      | 6.7.10-200      | 6.9.0-0.rc1.20240329git317c7bc0ef03.20 |
|               |                |                | 6.7.10-200&nbsp;&#x21D1; |                | 6.9.0-0.rc1.20240326git928a87efa423.17&nbsp;&#x21D1; |
|*grub2-common* | 2.06-13+deb12u1 | 2.06-116        | 2.06-118        | 2.06-118        | 2.06-119        |
|               |                |                |                |                |                |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-1        | 1.10.3-1        | 1.10.3-3        |
|               |                |                |                |                |                |
|*buildah*      | 1.33.7+ds1-1    | 1.35.1-1        | 1.35.2-1        | 1.35.0-1        | 1.35.2-2        |
|               |                |                |                |                | 1.35.0-1&nbsp;&#x21D1; |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       | 2.230.0-1       |
|               |                |                |                |                |                |
|*containers-common*| ?               | 1-89            | 1-99            | 1-99            | 0.58.0-5        |
|               |                |                |                |                |                |
|*criu*         | 3.17.1-3        | 3.18-1          | 3.19-2          | 3.19-2          | 3.19-4          |
|               |                |                |                |                |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-5        |
|               |                |                |                |                | 1.14.4-1&nbsp;&#x21D1; |
|*golang*       | 2:1.22~3        | 1.21.8-1        | 1.21.8-1        | 1.21.8-1        | 1.22.1-4        |
|               |                |                |                |                |                |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-1         | 0.7.3-1         | 0.7.3-2         |
|               |                |                |                |                |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.93-2          | 7.94-1          | 7.94-1          | 7.94-1          |
|               |                |                |                |                |                |
|*passt*        | 2024-03-26      | 2024-03-26      | 2024-03-26      | 2024-03-26      | 2024-03-26      |
|               |                |                |                |                |                |
|*podman*       | 4.9.4+ds1-1     | 4.9.3-2         | 4.9.4-1         | 4.9.3-1         | 5.0.0-1         |
|               | 4.9.3+ds1-1+b1&nbsp;&#x21D1; |                |                |                |                |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-1        | 1.1.12-1        | 1.1.12-3        |
|               |                |                |                |                |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        | 1.14.2-2        |
|               |                |                |                | 1.14.2-1&nbsp;&#x21D1; |                |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-1         | 1.2.2-1         | 1.2.2-2         |
|               |                |                |                |                |                |
|*systemd*      | 255.4-1+b1      | 253.17-1        | 254.10-1        | 254.10-1        | 255.4-1         |
|               |                |                |                |                |                |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.34-8          | 1.35-2          | 1.35-2          | 1.35-3          |
|               |                |                |                |                |                |

**Comment by edsantiago** [comment]
[package versions are unchanged. Re-push was to update commit message and to add a Windows comment]

**Comment by edsantiago** [comment]
Passes [podman](https://github.com/containers/podman/pull/22207) and [buildah](https://github.com/containers/buildah/pull/5470)

**Review by cevich — APPROVED** [review]
LGTM.  Please merge if appropriate.  Thanks for testing and for sticking with this.

**Comment by edsantiago** [comment]
And, back to draft. This must not merge until https://github.com/containers/podman/pull/22207 merges.

**Comment by Luap99** [comment]
@edsantiago Did you test this out on all the other repos as well? skopeo, common, image, storage...
I think this has the risk to break on other repos as well and catching it only on the next (maybe urgent) update would be painful.

**Comment by edsantiago** [comment]
I will start opening PRs now on other repos.

**Comment by edsantiago** [comment]
| repo | status |
| --- | --- |
| [podman](https://github.com/containers/podman/pull/22207) | pass |
| [buildah](https://github.com/containers/buildah/pull/5470) | pass |
| [skopeo](https://github.com/containers/skopeo/pull/2308) | pass (except for blue-robot) |
| [c-image](https://github.com/containers/image/pull/2384) | pass |
| [c-storage](https://github.com/containers/storage/pull/1911) | pass |
| [c-common](https://github.com/containers/common/pull/1965) | fails [lint](https://github.com/containers/common/actions/runs/8815743913/job/24198313641?pr=1965), for reasons unlikely to be my fault |
| [aardvark](https://github.com/containers/aardvark-dns/pull/454) | FAIL [integration](https://cirrus-ci.com/task/5119056867491840) - apparently a flake. Also fails all blue-robots |
| [netavark](https://github.com/containers/netavark/pull/966) | FAIL - some expected (nc bug), some not (integration) |

**Comment by Luap99** [comment]
Thanks @edsantiago, I comfortable merging then. I assume you want the podman PR merged first?

**Comment by cevich** [comment]
Thanks for all your efforts on this @edsantiago and @Luap99 :smiley:

**Pull request merged**
